sklearn/tree/tests/test_tree.py

```diff
@@ -2689,10 +2689,8 @@ def test_regression_tree_missing_values_toy(Tree, X, criterion):
     impurity = tree.tree_.impurity
     assert all(impurity >= 0), impurity.min()  # MSE should always be positive
 
-    # Note: the impurity matches after the first split only on greedy trees
-    if Tree is DecisionTreeRegressor:
```

Comment:
With the fix that impurities match even for ExtraTreeRegressor, so I removed the `if` clause

Reply:
Uhm this is actually something that we wonder with @adam2392 in his PR originally. So it might have been this hidden bug.